### PR TITLE
Update install script to refresh Path

### LINKED ISSUE #3425: ebpfapi.dll is not in path
Apparently, the tests are not picking up the ebpfapi.dll in the program files path.

              TODO: File a bug that ebpfapi.dll is not in path

_Originally posted by @saxena-anurag in https://github.com/microsoft/ebpf-for-windows/pull/3421#discussion_r1548805314_

### PR DESCRIPTION
**Pull request by mikeagun**
## Description

Updates Path after install so EbpfApi.dll can be found (instead of copying the dll to current directory). Fixes #3425

## Testing

Covered by existing tests.

## Documentation

N/A

## Installation

N/A

### CODE REVIEW

File: scripts/install_ebpf.psm1 [hunk 1]
```diff
@@ -296,11 +296,10 @@ function Install-eBPFComponents
...
 
-    # Copy ebpfapi.dll from installation directory to current directory.
-    # This ensures latest version of ebpfapi is used by the tests.
-    $ebpfApiDllPath = Join-Path $env:ProgramFiles "ebpf-for-windows\ebpfapi.dll"
-    Write-Log("Copying ebpfapi.dll from '$ebpfApiDllPath' to '$pwd'...")
-    Copy-Item -Path $ebpfApiDllPath -Destination $pwd -Force
+    # Refresh Path so EbpfApi.dll can be found.
+    $machinepath = [system.environment]::getenvironmentvariable("path", [system.environmentvariabletarget]::machine)
+    $userpath = [system.environment]::getenvironmentvariable("path", [system.environmentvariabletarget]::user)
+    $env:path = $machinepath + ";" + $userpath
```
Comment (mtfriesen) [inline scripts/install_ebpf.psm1:302]: I recently copied and pasted this pattern into a Developer PowerShell for VS2022 and found it broke all the tools implicitly added to its path. I wonder if there's anything we can do to preserve those paths?

Comment (mtfriesen) [reply scripts/install_ebpf.psm1:302]: Looks like it doesn't matter in this scenario, which is PS remoting into a VM.

**Review by mtfriesen — APPROVED** [review]

File: scripts/install_ebpf.psm1 [hunk 2]
```diff
@@ -296,11 +296,10 @@ function Install-eBPFComponents
     }
 
 
-    # Copy ebpfapi.dll from installation directory to current directory.
-    # This ensures latest version of ebpfapi is used by the tests.
-    $ebpfApiDllPath = Join-Path $env:ProgramFiles "ebpf-for-windows\ebpfapi.dll"
-    Write-Log("Copying ebpfapi.dll from '$ebpfApiDllPath' to '$pwd'...")
-    Copy-Item -Path $ebpfApiDllPath -Destination $pwd -Force
+    # Refresh Path so EbpfApi.dll can be found.
```
Comment (shpalani) [inline scripts/install_ebpf.psm1:299]: The ebpfapi.dll is placed in ProgramFiles. In your changes, I do not see the dll being fetched from the ProgramFiles. Can you please help clarify?

File: scripts/install_ebpf.psm1 [hunk 3]
```diff
@@ -296,11 +296,10 @@ function Install-eBPFComponents
...
+    $machinepath = [system.environment]::getenvironmentvariable("path", [system.environmentvariabletarget]::machine)
```
Comment (shpalani) [inline scripts/install_ebpf.psm1:300]: Is ebpfapi.dll different for debug and release?

File: scripts/install_ebpf.psm1 (hunk 2 above)
Comment (mikeagun) [reply scripts/install_ebpf.psm1:299]: The installer copies the dll over and adds the eBPF directory to the system Path, but that doesn't update Path in the current PowerShell session.

This change re-loads the system Path after running the installer so the eBPF Program Files directory gets included and the dll found.

Comment (shpalani) [reply scripts/install_ebpf.psm1:299]: Thanks  @mikeagun for the clarification. 
1. Does it require admin privilege? 
2. Does it also remove the system path at uninstall?

File: scripts/install_ebpf.psm1 [hunk 4]
```diff
@@ -296,11 +296,10 @@ function Install-eBPFComponents
...
+    $userpath = [system.environment]::getenvironmentvariable("path", [system.environmentvariabletarget]::user)
```
Comment (shpalani) [inline scripts/install_ebpf.psm1:301]: Can you check  non-null constraints for those variables ($userpath, $machinepath)

File: scripts/install_ebpf.psm1 (hunk 2 above)
Comment (mikeagun) [reply scripts/install_ebpf.psm1:299]: 1. No additional privileges needed -- this doesn't modify the system Path, just loads the changes the installer previously made.
2. I didn't add anything to modify Path in the current shell session at uninstall time, but the uninstaller already removes the eBPF directory from the system Path.

File: scripts/install_ebpf.psm1 (hunk 4 above)
Comment (mikeagun) [reply scripts/install_ebpf.psm1:301]: I'm not sure I understand. If the environment variables aren't set, then this code would behave the same as if they are set to an empty string, so it shouldn't cause any problems.

Comment (shpalani) [reply scripts/install_ebpf.psm1:301]: Previously, it was copied from Program Files., which ensured that ebpfapi.dll was present.  With the new changes, how do you ensure the DLL is present and mandatory, if the userpath is not set?

File: scripts/install_ebpf.psm1 (hunk 1 above)
Comment (saxena-anurag) [inline scripts/install_ebpf.psm1:302]: nit: should we append to the existing `$env.path` instead of replacing it with `$machinepath + ";" + $userpath`?

**Review by shpalani — APPROVED** [review]

File: scripts/install_ebpf.psm1 (hunk 1 above)
Comment (mikeagun) [reply scripts/install_ebpf.psm1:302]: I wasn't sure which is better.

If we don't make any manual changes to the path then this is more general -- if at some point the installation adds anything else to the system path this code will already catch all the changes.

On the other hand, currently `Program Files\ebpf-for-windows` is the only thing added to the path during installation, so I could just add that one directory to `$env:Path`, and then it wouldn't matter whether we make other path changes within the script.

**Review by saxena-anurag — APPROVED** [review]

**Pull request merged**